src/ast/astExtractor/method-extractor.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  CstNode,
+  DimsCtx,
+  FormalParameterCtx,
+  MethodDeclaratorCtx,
+  MethodModifierCtx,
+  UnannClassTypeCtx,
+  VariableDeclaratorIdCtx,
+} from "java-parser";
+
+import { BaseJavaCstVisitorWithDefaults } from "java-parser";
+import { MethodModifier, MethodBody, MethodDeclaration, Identifier, FormalParameter } from "../types/classes";
+
+export class MethodExtractor extends BaseJavaCstVisitorWithDefaults {
+  private stack: Array<string> = [];
+  private modifier: Array<MethodModifier>;
+  private identifier: Identifier;
+  private params: Array<FormalParameter>;
+  private body: MethodBody;
+
+  constructor() {
+    super();
+    this.stack = [];
+    this.modifier = [];
+    this.identifier = '';
+    this.params = [];
+    this.body = [];
+    this.validateVisitor();
+  }
+
+  private getAndPop() {
+    const res = this.stack.at(-1);
```

Comment:
<img width="993" alt="image" src="https://github.com/source-academy/java-slang/assets/110974917/3f09e677-86be-4f37-a8e9-958f213ae04f">

`js-slang` currently is not targeting es2022, and we may want to follow suit.

Reply:
Thanks for pointing this out, fixed.